src/library/scala/math/BigInt.scala

```diff
@@ -176,80 +257,174 @@ final class BigInt(val bigInteger: BigInteger)
 
   /** Compares this BigInt with the specified BigInt for equality.
    */
-  def equals (that: BigInt): Boolean = compare(that) == 0
+  def equals (that: BigInt): Boolean =
+    if (this.longEncoding) {
+      if (that.longEncoding) this._long == that._long else false
```

Comment:
`that.longEncoding && (this._long == that.long)`
reads a bit clearer if you ask me, but this is a nitpick.

Reply:
Changed to
```scala
    if (this.longEncoding)
      that.longEncoding && (this._long == that._long)
    else
      !that.longEncoding && (this._bigInteger == that._bigInteger)
```